src/webview/ConnectionPanel.ts

```diff
@@ -0,0 +1,322 @@
+    if (ConnectionPanel.currentPanel) {
+      ConnectionPanel.currentPanel._panel.reveal(column);
+      return;
+    }
+
+    // Otherwise, create a new panel.
+    let panel: vscode.WebviewPanel;
+    if (mode === PanelMode.Add) {
+      panel = vscode.window.createWebviewPanel(
+        ConnectionPanel.viewType,
+        'Add Connection',
+        column || vscode.ViewColumn.One,
+        getWebviewOptions(extensionUri),
+      );
+      ConnectionPanel.viewType = 'addConnection';
+    } else {
+      panel = vscode.window.createWebviewPanel(
+        ConnectionPanel.viewType,
+        'Edit Connection',
+        column || vscode.ViewColumn.One,
+        getWebviewOptions(extensionUri),
+      );
+      ConnectionPanel.viewType = 'editConnection';
+    }
+
+    ConnectionPanel.currentPanel = new ConnectionPanel(panel, extensionUri, mode);
+    ConnectionPanel.currentPanel._sidebar = sideBar;
+
+    let oldConnection =
+      ConnectionManager.connectionManager.getAllConnections()[oldConnectionName as string];
+
+    ConnectionPanel.currentPanel._panel.webview.postMessage({
+      type: 'Connection.InitializeView',
+      connection: oldConnection,
+    });
+  }
+
+  public static revive(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, mode: PanelMode) {
+    ConnectionPanel.currentPanel = new ConnectionPanel(panel, extensionUri, mode);
+  }
+
+  private constructor(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, mode: PanelMode) {
+    this._panel = panel;
+    this._extensionUri = extensionUri;
+
+    // Set the webview's initial html content
+    this._update(mode);
+
+    // Listen for when the panel is disposed
+    // This happens when the user closes the panel or when the panel is closed programmatically
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+
+    // Update the content based on view changes
+    this._panel.onDidChangeViewState(
+      e => {
+        if (this._panel.visible) {
+          this._update(mode);
+        }
+      },
+      null,
+      this._disposables,
+    );
+
+    // Handle messages from the webview
+    this._panel.webview.onDidReceiveMessage(
+      message => {
+        switch (message.type) {
+          case 'Connection.cancel': {
+            this.dispose();
+            break;
+          }
+          case 'Connection.testConnection': {
+            const panel = vscode.window.createWebviewPanel(
+              'catCoding',
+              'Cat Coding',
+              ViewColumn.One,
+              {},
+            );
+
+            panel.webview.html = `<!DOCTYPE html>
+              <html lang="en">
+              <head>
+                  <meta charset="UTF-8">
+                  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+                  <title>Cat Coding</title>
+              </head>
+              <body>
+                  <img src="https://media.giphy.com/media/JIX9t2j0ZTN9S/giphy.gif" width="300" />
```

Comment:
This is cute, and in a personal project I'd do the same, but I do worry about the case where something like this accidentally makes it into something we officially distribute (it's been known to happen!)